utils.js

```diff
@@ -1,23 +1,37 @@
+function getTrailerID() { // Requires DOM level access
+    let p = document.location.pathname;
+    if(!document.location.origin === "https://www.youtube.com") return false
+    if(!p.startsWith("/user/")  && !p.startsWith("/channel/"))  return false
+    if(document.getElementsByClassName("ytp-title-link")[1] === undefined) return false
+    let id = document.getElementsByClassName("ytp-title-link")[1].href.split("?v=")[1];
+    return id.length == 11 ? id : false;
+}
+
 function getYouTubeVideoID(url) {
+    if(url === undefined) {
+        if(id = getTrailerID()) return id;
+        url = document.URL;
+    }
     //Attempt to parse url
-    let urlObject = null;
-    try { 
-        urlObject = new URL(url);
-    } catch (e) {      
-        console.error("[SB] Unable to parse URL: " + url);
-        return false
+    if(typeof url === 'string') {
```

Comment:
What happens if this is false?

Reply:
the parsing will be skipped this was done to allow url objects to be parsed to it directly but never got used

Reply:
Removed check